src/software/ai/hl/stp/play/ball_placement_play_test.cpp

```diff
@@ -0,0 +1,46 @@
+#include "software/ai/hl/stp/play/ball_placement_play.h"
+
+#include <gtest/gtest.h>
+
+#include "software/simulated_tests/simulated_play_test_fixture.h"
+#include "software/simulated_tests/terminating_validation_functions/ball_at_point_validation.h"
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/test_util/test_util.h"
+#include "software/time/duration.h"
+#include "software/world/world.h"
+
+class BallPlacementPlayTest : public SimulatedPlayTestFixture
+{
+};
+
+TEST_F(BallPlacementPlayTest, test_ball_placement)
+{
+    Point ball_placement_point(-3, -2);
+    setBallState(BallState(Point(0, 0.5), Vector(0, 0)));
+    addFriendlyRobots(TestUtil::createStationaryRobotStatesWithId(
+        {Point(-3, 2.5), Point(-3, 1.5), Point(-3, 0.5), Point(-3, -0.5), Point(-3, -1.5),
+         Point(4.6, -3.1)}));
+    setFriendlyGoalie(0);
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId(
+        {Point(1, 0), Point(1, 2.5), Point(1, -2.5), field().enemyGoalCenter(),
+         field().enemyDefenseArea().negXNegYCorner(),
+         field().enemyDefenseArea().negXPosYCorner()}));
+    setEnemyGoalie(0);
+    setAIPlay(TYPENAME(BallPlacementPlay));
+    GameState game_state;
+    game_state.updateRefereeCommand(RefereeCommand::HALT);
+    game_state.updateRefereeCommand(RefereeCommand::BALL_PLACEMENT_US);
+    game_state.setBallPlacementPoint(ball_placement_point);
+    setGameState(game_state);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [this, ball_placement_point](std::shared_ptr<World> world_ptr,
+                                     ValidationCoroutine::push_type& yield) {
+            ballAtPoint(ball_placement_point, world_ptr, yield);
```

Comment:
In the code it says 
```
 // non goalie and non ball placing robots line up along a line just outside the
 // friendly defense area to wait for ball placement to finish
```

Should we also test for that?

Reply:
This is an implementation detail that shouldn't be tested. Since it's friendly ball placement, it doesn't matter what the other robots do as long as the ball gets placed in the right spot.